### ISSUE TITLE
Web Fonts Performance: Making Pretty, Fast

### ISSUE THREAD

**Issue by marcobiedermann** [stefanjudis/perf-tooling#458]
https://www.igvita.com/2012/09/12/web-fonts-performance-making-pretty-fast/

**Comment by marcobiedermann** [comment]
To be honest, I am a bit confused about Webfont served by Google Webfont.
In their article they say, that the font will be cached for one year and if the font was already being downloaded form another site, which uses the same font, it load up from cache.
The filementgroup points out that you shout not use Google Webfonts and serve the fonts by yourself in base64 encoded.
So which one would be better?
I personally think that serving the fonts by yourself is better because every time I measure a page using Google Pagespeed Insight, I get a bad reporting because of uncached webfonts from Google.

@stefanjudis What is your opinion on this topic?
Because the article is a bit old and we have some better articles already listed, I would give it a :-1:

**Comment by stefanjudis** [comment]
maybe @igrigorik can give also some hint here - if he still think his article is valid.

I'm not that much into fonts - but can check out in detail and play around with some cases. :)

**Comment by igrigorik** [comment]
See: https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/webfont-optimization?hl=en

**Comment by marcobiedermann** [comment]
@igrigorik Thank you very much for the article.
So to sum it up, use only Google Webfonts if you want to quickly try out some fonts or prefer simplicity over performance.
If you are really going for performance optimized website, you should not use any external webfont service and provide them by yourself and load them async using the font API.
Right? :)

**Comment by igrigorik** [comment]
> So to sum it up, use only Google Webfonts if you want to quickly try out some fonts or prefer simplicity over performance.

No. This is a long discussion, but tl;dr: yes, there is a tradeoff with extra CSS request, but that request is also what enables Google Fonts to deliver an optimized experience for various browsers. E.g. googledevelopers.blogspot.com/2015/02/smaller-fonts-with-woff-20-and-unicode.html. 

Unless you have a strict requirement for self-hosting fonts, or are willing to invest a lot of time into optimizing your font-serving, I recommend you stick with Google Fonts.

**Comment by marcobiedermann** [comment]
Ok, thanks for the clarification :)
